File: Posicion_Redireccionamiento.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
    int n,m,n2,m2;
    int matriz[m][n];
    int var;
    int idx;
    int i=0;
    int j=0;
    int dir_matriz;
    int n_ren;
    int tam=4;
    int dir_celda;
    
    printf("\tTamaño de Columnas:");
    scanf("%d",&m);
    printf("\n\tTamaño de filas:");
    scanf("%d",&n);
    
    
    while(1){
        printf("\n\tInicio");
        printf("\n\t1.-Polinomio de Direccionamiento\n\ty\n\tDireccion de la celda\n\t2.-Salir\n");
        scanf("%d",&var);
        switch(var){
            case 1:
            printf("\n\tDigita la ubicacion que quieres obtener");
            printf("\n[");
            scanf("%d",&m2);
            printf("][");
            scanf("%d",&n2);
            printf("]");
            idx=m2*n+n2;
            printf("\n\tIndica la direccion de la matriz: ");
            scanf("%d",&dir_matriz);
            dir_celda=(dir_matriz+(m*m2+n2))*tam;
            printf("\n\tLa Indice de Direccionamiento es: %i",idx);
            printf("\n\tDireccion de la celda especificadad: %i",dir_celda);
            
            break;
            case 2:
            printf("\n\tBuen dia");
            exit(0);
            break;
            default:
            printf("\n\tOPCION NO VALIDA");
            break;
        }
        
    }
    

    return 0;
}

File: Posicion_redireccionamiento.py
m=int(input("\tTamaño de Columnas: "))
n=int(input("\n\tTamaño de Filas: "))
tam=4
var=1

while var!=2:
    print("\n\tInicio")
    print("\n\t1.-Polinomio de Direccionamienyo\n\tY\n\tDireccion de Celda\n\t2.-Salir")
    var=int(input(""))
    if (var==1):
        m2=int(input("\n\tDigita la ubicacion que Quieres obtener:\n ["))
        print("][")
        n2=int(input(""))
        print("]")
        idx=(m2*n)+n2
        dir_matriz=int(input("\n\tIndica la direccion de la matriz: "))
        dir_celda=(((dir_matriz+((m*m2)+n2)))*tam)
        print("\n\tEl indice de direccionamiento es: ")
        print(idx)
        print("\n\tDireccion de la celda especificada: ")
        print(dir_celda)
    else:
        if (var==2):
            print("\n\tBuen dia")
        else: 
            if (var<1 or var>2):
                print("\n\tOpcion no valida")